### DISCUSSION TITLE
Go a-h templ

### DISCUSSION THREAD

**Discussion by batyrmuhadov** [LazyVim/LazyVim#3735] [Q&A] [answered]
How to config a-h templ lsp ???

**Comment by derpycoder** [comment]
I need some guidance as well.

Did you figure it out @batyrmuhadov ?

**Reply by batyrmuhadov** [reply]
Yes, look at the lsp docs on the official website

**Reply by derpycoder** [reply]
Can you please write a snippet to guide me?

I went through the docs, but I got confused and my attempts failed.

```lua
setup = {
        -- example to setup with typescript.nvim
        -- tsserver = function(_, opts)
        --   require("typescript").setup({ server = opts })
        --   return true
        -- end,
        -- Specify * to use this function as a fallback for any server
        -- ["*"] = function(server, opts) end,
      },
```

I tried adding 

```lua
templ = function(_, opts) 
require(“templ”).setup({})
end
```

And in another file, I imported templ.


However it didn’t work.

Can you please post whatever files and related changes are necessary to get templ working?

**Reply by batyrmuhadov** [reply]
create templ.lua and there add something like this
```
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        templ = {
          filetypes = { "templ"},
          settings = {
            templ = {
              enable_snippets = true,
            },
          },
        },
      },
    },
  },
}
```

**Reply by derpycoder** [reply]
It didn't work.

![image](https://github.com/user-attachments/assets/651ff181-c709-4393-ba1b-b18fec712c5a)
![image](https://github.com/user-attachments/assets/6c0f6184-13e3-4ee6-badc-792c192a70da)
![image](https://github.com/user-attachments/assets/fb3db767-b310-4b5d-97f4-c08ef3d0df65)
![image](https://github.com/user-attachments/assets/613e8d3a-e33a-40e8-b75f-111d4dbca88a)

I created these files. However I am still getting this error.
![image](https://github.com/user-attachments/assets/398e7ebd-54e1-4ae7-a4dd-77cb12e620aa)

templ is installed
![image](https://github.com/user-attachments/assets/0afadddf-1da7-4f82-b789-05a98926ad15)

**Reply by derpycoder** [reply]
I also added a conform file for formatting, which works

![image](https://github.com/user-attachments/assets/b7f6c522-9541-4f06-ab32-edb6f1c16791)

I am also correctly getting Tailwindcss autocompletion.

**Reply by derpycoder** [reply]
@batyrmuhadov,

Can you share all the files where you made changes to get templ to work.

or perhaps your dotfile repo?

**Reply by dpetka2001** [reply]
You have an error in `/lua/plugins/lsp.lua` on line 72. You should not have to define anything in the `setup` field. Remove lines 71-73 completely.

**Reply by derpycoder** [reply]
I removed them, but I still don't see syntax highlighting, but now at least the code completions seems to work.

**Reply by dpetka2001** [reply]
Syntax highlighting comes from Treesitter. Make sure you have the necessary parser installed.

**Reply by derpycoder** [reply, marked as answer]
I got it to work. I will post screenshots.

Thank you, @dpetka2001 & @batyrmuhadov, for your guidance.

---
Here's how it looks:
![image](https://github.com/user-attachments/assets/b7541462-39f2-4b48-a1b4-40abbf6328ee)

---
## Here are the files that you need to add to make the above work:
### Mason, To fetch LSP
`~/.config/nvim/lua/plugins/mason.lua`
```lua
return {
  "williamboman/mason.nvim",
  event = "LazyFile",
  cmd = "Mason",
  keys = { { "<leader>cm", "<cmd>Mason<cr>", desc = "Mason" } },
  build = ":MasonUpdate",
  opts_extend = { "ensure_installed" },
  opts = {
    ensure_installed = {
      "templ",
    },
  },
}
```
---
### Configure the `templ` LSP
`~/.config/nvim/lua/plugins/templ.lua`
```lua
return {
  "neovim/nvim-lspconfig",
  event = "LazyFile",
  opts = {
    servers = {
      templ = {
        filetypes = { "templ" },
        settings = {
          templ = {
            enable_snippets = true,
          },
        },
      },
    },
  },
}
```
---
### Treesitter, for syntax highlighting
`~/.config/nvim/lua/plugins/treesitter.lua`
```lua
return {
  "nvim-treesitter/nvim-treesitter",
  event = { "LazyFile", "VeryLazy" },
  opts_extend = { "ensure_installed" },
  opts = {
    highlight = { enable = true },
    indent = { enable = true },
    ensure_installed = {
      "bash",
      "c",
      "diff",
      "html",
      "javascript",
      "jsdoc",
      "json",
      "jsonc",
      "lua",
      "luadoc",
      "luap",
      "markdown",
      "markdown_inline",
      "printf",
      "query",
      "toml",
      "vim",
      "vimdoc",
      "yaml",
      "templ",
    },
  },
}
```
---
### Conform, for formatting the `templ` stuff
`~/.config/nvim/lua/plugins/conform.lua`
```lua
return {
  "stevearc/conform.nvim",
  opts = {
    formatters_by_ft = {
      templ = { "templ" },
    },
  },
}
```
---
That's all.
